File: ios-native/IronCore/Models/Meal.swift
import Foundation
import FirebaseFirestore

/// Meal log at `users/{userId}/meals/{mealId}`
struct Meal: Codable, Identifiable {
    @DocumentID var id: String?
    var date: String            // "YYYY-MM-DD"
    var userId: String
    var mealName: String
    var calories: Double
    var protein: Double
    var carbs: Double
    var fat: Double
    @ServerTimestamp var createdAt: Timestamp?
}

File: android-native/app/src/main/java/com/ironcore/fit/data/model/Meal.kt
package com.ironcore.fit.data.model

import com.google.firebase.Timestamp
import com.google.firebase.firestore.ServerTimestamp

/**
 * Firestore path: /users/{userId}/meals/{mealId}
 *
 * Single source of truth for meal data. Used by NutritionRepository,
 * NutritionViewModel, HomeViewModel, HomeScreen, and NutritionDay.
 */
data class Meal(
    val id: String = "",
    val name: String = "",
    val calories: Int = 0,
    val protein: Double = 0.0,
    val carbs: Double = 0.0,
    val fat: Double = 0.0,
    val time: Timestamp? = null,
    val date: String = "",
    val aiAnalyzed: Boolean = false,
    val userId: String = "",
    val notes: String = "",
    @ServerTimestamp val createdAt: Timestamp? = null
)
